Add transformers MMS checkpoints to docs (#5186)
facebookresearch/fairseq: examples/mms/README.md

**🤗 Transformers**

First, we install transformers and some other libraries
```
pip install torch accelerate torchaudio datasets
pip install --upgrade transformers
````

**Note**: In order to use MMS you need to have at least `transformers >= 4.30` installed. If the `4.30` version
is not yet available [on PyPI](https://pypi.org/project/transformers/) make sure to install `transformers` from 
source:
```
pip install git+https://github.com/huggingface/transformers.git
```

Next, we load a couple of audio samples via `datasets`. Make sure that the audio data is sampled to 16000 kHz.

```py
from datasets import load_dataset, Audio

# English
stream_data = load_dataset("mozilla-foundation/common_voice_13_0", "en", split="test", streaming=True)
stream_data = stream_data.cast_column("audio", Audio(sampling_rate=16000))
en_sample = next(iter(stream_data))["audio"]["array"]

> # French
> stream_data = load_dataset("mozilla-foundation/common_voice_13_0", "fr", split="test", streaming=True)
> stream_data = stream_data.cast_column("audio", Audio(sampling_rate=16000))
> fr_sample = next(iter(stream_data))["audio"]["array"]

Author: ```suggestion
# Swahili
stream_data = load_dataset("mozilla-foundation/common_voice_13_0", "sw, split="test", streaming=True)
stream_data = stream_data.cast_column("audio", Audio(sampling_rate=16000))
sw_sample = next(iter(stream_data))["audio"]["array"]
```

```

Next, we load the model and processor

```py
from transformers import Wav2Vec2ForCTC, AutoProcessor
import torch

model_id = "facebook/mms-1b-all"

processor = AutoProcessor.from_pretrained(model_id)
model = Wav2Vec2ForCTC.from_pretrained(model_id)
```

Now we process the audio data, pass the processed audio data to the model and transcribe the model output, just like we usually do for Wav2Vec2 models such as [facebook/wav2vec2-base-960h](https://huggingface.co/facebook/wav2vec2-base-960h)

```py
inputs = processor(en_sample, sampling_rate=16_000, return_tensors="pt")

with torch.no_grad():
    outputs = model(**inputs).logits

ids = torch.argmax(outputs, dim=-1)[0]
transcription = processor.decode(ids)
# 'joe keton disapproved of films and buster also had reservations about the media'
```

We can now keep the same model in memory and simply switch out the language adapters by calling the convenient [`load_adapter()`]() function for the model and [`set_target_lang()`]() for the tokenizer. We pass the target language as an input - "fra" for French.

```py

> processor.tokenizer.set_target_lang("fra")
> model.load_adapter("fra")
> 
> inputs = processor(fr_sample, sampling_rate=16_000, return_tensors="pt")
> 
> with torch.no_grad():
>     outputs = model(**inputs).logits
> 
> ids = torch.argmax(outputs, dim=-1)[0]
> transcription = processor.decode(ids)
> # "ce dernier est volé tout au long de l'histoire romaine"

Author: ```suggestion
processor.tokenizer.set_target_lang("swh")
model.load_adapter("swh")

inputs = processor(sw_sample, sampling_rate=16_000, return_tensors="pt")

with torch.no_grad():
    outputs = model(**inputs).logits

ids = torch.argmax(outputs, dim=-1)[0]
transcription = processor.decode(ids)
# 'wachambuzi wa soka wanamtaja mesi kama nyota hatari zaidi duniani'
# => In English: "soccer analysts describe Messi as the most dangerous star in the world"
```

\* In the `Dictionary` column, we provide the download link for token dictionary in English language. To download token dictionary for a different language supported by the model, modify the language code in the URL appropriately. For example, to get token dictionary of FL102 model for Hindi language, use [this](https://dl.fbaipublicfiles.com/mms/asr/dict/mms1b_fl102/hin.txt) link. 

**🤗 Transformers**

First, we install transformers and some other libraries
```

> pip install torch accelerate torchaudio datasets

Reviewer: As far as I'm aware, we don't need `accelerate` or `torchaudio` no? Note that all audio decoding / resampling is done with librosa/soundfile in datasets now, both of which can be installed with:

```suggestion
pip install torch datasets[audio]
```

pip install --upgrade transformers
````

**Note**: In order to use MMS you need to have at least `transformers >= 4.30` installed. If the `4.30` version
is not yet available [on PyPI](https://pypi.org/project/transformers/) make sure to install `transformers` from 
source:
```
pip install git+https://github.com/huggingface/transformers.git
```

Next, we load a couple of audio samples via `datasets`. Make sure that the audio data is sampled to 16000 kHz.

```py
from datasets import load_dataset, Audio

# English

> stream_data = load_dataset("mozilla-foundation/common_voice_13_0", "en", split="test", streaming=True)

Reviewer: Note that this will only be allowed if the user has accepted the Mozilla terms of use on the Hub and linked their HF account - perhaps we can use an un-gated dataset?

stream_data = stream_data.cast_column("audio", Audio(sampling_rate=16000))
en_sample = next(iter(stream_data))["audio"]["array"]

# Swahili

> stream_data = load_dataset("mozilla-foundation/common_voice_13_0", "sw, split="test", streaming=True)

Reviewer: ```suggestion
stream_data = load_dataset("mozilla-foundation/common_voice_13_0", "sw", split="test", streaming=True)
```

stream_data = stream_data.cast_column("audio", Audio(sampling_rate=16000))
sw_sample = next(iter(stream_data))["audio"]["array"]
```

Next, we load the model and processor

```py
from transformers import Wav2Vec2ForCTC, AutoProcessor
import torch

model_id = "facebook/mms-1b-all"

processor = AutoProcessor.from_pretrained(model_id)
model = Wav2Vec2ForCTC.from_pretrained(model_id)
```

Now we process the audio data, pass the processed audio data to the model and transcribe the model output, just like we usually do for Wav2Vec2 models such as [facebook/wav2vec2-base-960h](https://huggingface.co/facebook/wav2vec2-base-960h)

```py
inputs = processor(en_sample, sampling_rate=16_000, return_tensors="pt")

with torch.no_grad():
    outputs = model(**inputs).logits

ids = torch.argmax(outputs, dim=-1)[0]
transcription = processor.decode(ids)
# 'joe keton disapproved of films and buster also had reservations about the media'
```

> We can now keep the same model in memory and simply switch out the language adapters by calling the convenient [`load_adapter()`]() function for the model and [`set_target_lang()`]() for the tokenizer. We pass the target language as an input - "fra" for French.

Reviewer: Shall we link the docs to `load_adapter` and `set_target_lang` here?

Reviewer: ```suggestion
We can now keep the same model in memory and simply switch out the language adapters by calling the convenient [`load_adapter()`]() function for the model and [`set_target_lang()`]() for the tokenizer. We pass the target language as an input - "swh" for Swahili.
```

```py
processor.tokenizer.set_target_lang("swh")
model.load_adapter("swh")

inputs = processor(sw_sample, sampling_rate=16_000, return_tensors="pt")

with torch.no_grad():
    outputs = model(**inputs).logits

ids = torch.argmax(outputs, dim=-1)[0]
transcription = processor.decode(ids)
# 'wachambuzi wa soka wanamtaja mesi kama nyota hatari zaidi duniani'

> # => In English: "soccer analysts describe Messi as the most dangerous star in the world"

Author: ```suggestion
# => In English: "soccer analysts describe Messi as the most dangerous player in the world"
```
